snowshu/adapters/target_adapters/base_remote_target_adapter.py

```diff
@@ -1,5 +1,24 @@
+import logging
+from io import StringIO
+from typing import Union
+
+from snowshu.core.configuration_parser import ConfigurationParser, TEMPLATES_PATH
```

Comment:
Do we need to reference the config parser here? Would it make more sense to have the replica factory pass in the credentials after they are run through the config parser?

Reply:
Well, current state of getting credentials is happening in ConfigurationParser().from_file_or_path() and unfortunately it's different for source and target adapters. Source is doing it while running _build_adapter_profile() (assumes it's always remote source) based on passed credpath, however target is using a custom method (currently the one defined here and the other defined in BaseLocalTargetAdapter and always assumes local target), where the local is having that assumption that creds are always "snowshu". I think we could scope out a ticket to refactor it and possibly contain just one method for creating adapters (not two as we have now) and then wrap somehow that if target credentials are now passed we set "snowshu" as default ones, or something similar for simplicity. Unless I go to far with it and you see a simpler solution?

Reply:
Yeah, that makes sense to me since locally we don't worry about non-default credentials.

I'd leave it mostly as is for now, but let's separate the work out for sometime later to rework the config and credentials handling. My only quick suggestion would be moving the global var for the schema json file to the config parser.

Reply:
https://health-union.atlassian.net/browse/DATA-7221